src/Extension/Propagator/Jaeger/JaegerPropagator.php

```diff
@@ -0,0 +1,174 @@
+ * "uber-trace-id" used for trace context propagation across service boundaries.
+ * (https://www.jaegertracing.io/docs/1.52/client-libraries/#propagation-format)
+ */
+class JaegerPropagator implements TextMapPropagatorInterface
+{
+    private const UBER_TRACE_ID_HEADER = 'uber-trace-id';
+    private const UBER_BAGGAGE_HEADER_PREFIX = 'uberctx';
+
+    private const IS_NOT_SAMPLED = '0';
+    private const IS_SAMPLED = '1';
+    private const IS_DEBUG = '2';
+    private const DEFAULT_PARENT_SPAN_ID = '0';
+
+    private const FIELDS = [
+        self::UBER_TRACE_ID_HEADER,
+    ];
+
+    private static ?TextMapPropagatorInterface $instance = null;
+
+    public static function getInstance(): TextMapPropagatorInterface
+    {
+        if (self::$instance === null) {
+            self::$instance = new JaegerPropagator();
+        }
+
+        return self::$instance;
+    }
+
+    public function fields(): array
+    {
+        return self::FIELDS;
+    }
+
+    /** {@inheritdoc} */
+    public function inject(&$carrier, PropagationSetterInterface $setter = null, ContextInterface $context = null): void
+    {
+        $setter ??= ArrayAccessGetterSetter::getInstance();
+        $context ??= Context::getCurrent();
+        $spanContext = Span::fromContext($context)->getContext();
+
+        if (!$spanContext->isValid()) {
+            return;
+        }
+
+        $flag = $this->getFlag($spanContext, $context);
+
+        $uberTraceId = sprintf(
+            '%s:%s:%s:%s',
+            $spanContext->getTraceId(),
+            $spanContext->getSpanId(),
+            self::DEFAULT_PARENT_SPAN_ID,
+            $flag
+        );
+
+        $setter->set($carrier, self::UBER_TRACE_ID_HEADER, $uberTraceId);
+
+        $baggage = Baggage::fromContext($context);
+
+        if ($baggage->isEmpty()) {
+            return;
+        }
+
+        /** @var Entry $entry */
+        foreach ($baggage->getAll() as $key => $entry) {
+            $value = rawurlencode((string) $entry->getValue());
+            $setter->set(
+                $carrier,
+                sprintf('%s-%s', self::UBER_BAGGAGE_HEADER_PREFIX, $key),
+                $value
+            );
+        }
+    }
+
+    /** {@inheritdoc} */
+    public function extract($carrier, PropagationGetterInterface $getter = null, ContextInterface $context = null): ContextInterface
+    {
+        $getter ??= ArrayAccessGetterSetter::getInstance();
+        $context ??= Context::getCurrent();
+
+        $spanContext = self::extractImpl($carrier, $getter, $context);
+        if (!$spanContext->isValid()) {
+            return $context;
+        }
+
+        return $context->withContextValue(Span::wrap($spanContext));
+    }
+
+    private function getFlag(SpanContextInterface $spanContext, ContextInterface $context): string
+    {
+        if ($spanContext->isSampled()) {
+            if ($context->get(JaegerDebugFlagContextKey::instance())) {
+                return self::IS_DEBUG;
```

Comment:
Sampled + Debug = 3

Reply:
So it would be something like:

`return self::IS_DEBUG | self::IS_SAMPLED`